docs/src/advanced-concepts/preparing-generic-targets.rst

```diff
@@ -0,0 +1,110 @@
+Preparing generic targets for reading by metatrain
+==================================================
+
+Besides energy-like targets, the library also supports reading (and training on)
+more generic targets.
+
+Input file
+----------
+
+In order to read a generic target, you will have to specify its layout in the input
+file. Suppose you want to learn a target named ``mtt::my_target``, which is a per-atom
+Cartesian vector with 10 properties. The target section in the input file should look
+like this:
+
+.. code-block:: yaml
+
+    targets:
+      mtt::my_target:
+        quantity: ""
+        read_from: dataset.xyz
+        key: energy
+        unit: ""
+        per_atom: True
+        type:
+          cartiesian:
+            rank: 1
+        num_properties: 10
```

Comment:
This could also be called `n_targets`/`num_targets` to clarify that the "properties" are fully independent from one another.

Reply:
I don't really love `num_targets: 10` here because the total number is "kinda 30"

Reply:
It's 10 independent targets each with 3 vector components. But we can find another name!

Reply:
I've improved the docs